data/basics/userBalance.go

```diff
@@ -96,15 +96,29 @@ func UnmarshalStatus(value string) (s Status, err error) {
 	return
 }
 
+// AgreementAccountData contains just the account data used by agreement.
+type AgreementAccountData struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Status     Status     `codec:"onl"`
+	MicroAlgos MicroAlgos `codec:"algo"`
+
+	VoteID      crypto.OneTimeSignatureVerifier `codec:"vote"`
+	SelectionID crypto.VRFVerifier              `codec:"sel"`
+
+	VoteFirstValid  Round  `codec:"voteFst"`
+	VoteLastValid   Round  `codec:"voteLst"`
+	VoteKeyDilution uint64 `codec:"voteKD"`
+}
+
 // AccountData contains the data associated with a given address.
 //
 // This includes the account balance, cryptographic public keys,
 // consensus delegation status, asset data, and application data.
 type AccountData struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
-	Status     Status     `codec:"onl"`
-	MicroAlgos MicroAlgos `codec:"algo"`
+	AgreementAccountData
```

Comment:
Does an anonymous embedding of a struct like this impact the msgp encoding? That would break catchpoints ... I didn't check originally. If so (or just to be safe, maybe it's best) we could have AgreementAccountData and AccountData be two completely separate types, and then have a copy constructor convert an AccountData to AgreementAccountData in LookupAgreement. Maybe the embedding was too cute and possibly disruptive (changing all the places where tests declare AccountData structs etc)